### PR TITLE
Flamegraph scripts

### PR DESCRIPTION
**Pull request by oazizi000**

### CODE REVIEW

File: pxl_scripts/px/perf_flamegraph/trace.pxl [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Pixie Labs, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License")
+
+'''PEM Flamegraph
```
Comment (htroisi) [inline pxl_scripts/px/perf_flamegraph/trace.pxl:4]: CPU Flamegraph?

File: pxl_scripts/px/perf_flamegraph/vis.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  ],
+  "globalFuncs": [],
+  "widgets": [
+    {
+      "name": "Flamegraph",
+      "position": {
+        "x": 0,
+        "y": 0,
+        "w": 10,
```
Comment (htroisi) [inline pxl_scripts/px/perf_flamegraph/vis.json:46]: Why not full width (12)?

File: pxl_scripts/px/perf_flamegraph/vis.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      "defaultValue": "-5m"
+    },
+    {
+      "name": "node",
+      "type": "PX_NODE",
+      "description": "The node to filter on.",
+      "defaultValue": ""
+    },
+    {
+      "name": "namespace",
```
Comment (htroisi) [inline pxl_scripts/px/perf_flamegraph/vis.json:16]: I'd switch the order of the arguments: namespace > node > pod makes more sense.

File: pxl_scripts/px/perf_flamegraph/trace.pxl [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    node_agg = df.groupby([pct_grouping_entity]).agg(
+        count=('count', px.sum)
+    )
+
+    # Apply filters.
+    df = df[px.contains(df.node, node)]
+    df = df[px.contains(df.pod, pod)]
+    df = df[px.contains(df.namespace, namespace)]
+
+    # Aggregate across time windows.
```
Comment (htroisi) [inline pxl_scripts/px/perf_flamegraph/trace.pxl:32]: IIUC, there aren't really time windows, just samples with timestamps. Maybe "Aggregate across all samples?"

File: pxl_scripts/px/perf_flamegraph/vis.json (hunk 3 above)
Comment (oazizi000) [reply pxl_scripts/px/perf_flamegraph/vis.json:16]: Acutally, node is independent of namespace/pod/container/pid.

For example, a node can contain multiple namespaces. Also, a namespace can span multiple nodes. So it's really an any-to-any relationship, not a parent-child relationship.

So we need to keep the namespace and pod filters adjacent to each other. Those two do have a parent-child relationship.

File: pxl_scripts/px/perf_flamegraph/trace.pxl (hunk 4 above)
Comment (oazizi000) [reply pxl_scripts/px/perf_flamegraph/trace.pxl:32]: These are actually aggregating different flamegraphs into each other. For example, we dump a flamegraph every 30 seconds for the last 30-second time window. But if we query for the last five minutes, it'll take the ten flamegraphs and merge them into one larger flamegraph that covers the entire ten minute time window.

I'll update the comment to make that a bit clearer.

File: pxl_scripts/px/perf_flamegraph/vis.json (hunk 3 above)
Comment (htroisi) [reply pxl_scripts/px/perf_flamegraph/vis.json:16]: Oh, whoops. I'm not sure what I was thinking. :)

**Review by htroisi — APPROVED** [review]

**Review by zasgar — APPROVED** [review]

**Pull request merged**